packages/scoped-elements/README.md

```diff
@@ -166,6 +85,141 @@ export class MyElement extends ScopedElementsMixin(LitElement) {
 }
 ```
 
+### Lazy scoped components
+
+In some situations may happen that you want to use a component in your templates that is not already loaded at the moment of defining the scoped elements map. The `ScopedElementsMixin` provides the `defineScopedElement` method to define scoped elements at any time.
+
+```js
+import { LitElement } from 'lit-element';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements';
+import { MyPanel } from './MyPanel.js';
+
+export class MyElement extends ScopedElementsMixin(LitElement) {
+  static get scopedElements() {
+    return {
+      'my-panel': MyPanel,
+    };
+  }
+
+  constructor() {
+    super();
+
+    import('./MyButton.js').then(({ MyButton }) => this.defineScopedElement('my-button', MyButton));
+  }
+
+  render() {
+    return html`
+      <my-panel class="panel">
+        <my-button>${this.text}</my-button>
+      </my-panel>
+    `;
+  }
+}
+```
+
+### Obtaining a scoped tag name
+
+Maybe you want to create a scoped element programmatically and don't know which one is the scoped tag name? No problem, there is a static method called `getScopedTagName` that would help you for that.
+
+```js
+import { LitElement } from 'lit-element';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements';
+import { MyButton } from './MyButton.js';
+import { MyPanel } from './MyPanel.js';
+
+export class MyElement extends ScopedElementsMixin(LitElement) {
+  static get scopedElements() {
+    return {
+      'my-panel': MyPanel,
+      'my-button': MyButton,
+    };
+  }
+
+  constructor() {
+    super();
+
+    const scopedTagName = this.constructor.getScopedTagName('my-button');
+
+    // do whatever you need with the scopedTagName
+  }
+
+  // ...
+}
+```
+
+## Motivation
+
+Complex Web Component applications are often developed by several teams across organizations. In that scenario it is common that shared component libraries are used by teams to create a homogeneous look and feel or just to avoid creating the same components multiple times, but as those libraries evolve problems between different versions of the same library may appear, as teams may not be able to evolve and update their code at the same velocity. This causes bottlenecks in software delivery that should be managed by the teams and complex build systems, to try to alleviate the problem.
```

Comment:
@LarsDenBakker I didn't removed the text, I just moved it here, after the API docs and I title it as Motivation.

I removed `How to install` as we don't have it in any other docs.

Reply:
A small explanation of what the library does I think is good as a start, but indeed we should move most of the in-depth explanation to the bottom.

Reply:
Ok. I just restored the installation and I add a little explanation at the top. Let me know if it's better now.

thanks!